packages/webpack-plugin/src/mini-css-support.ts

```diff
@@ -46,8 +46,26 @@ export function injectCssModules(
                 }),
                 media: '',
                 sourceMap: null,
+                assets: {},
+                assetsInfo: {},
             });
 
+            try {
+                let syncCheck = false;
+                cssModule.build(
+                    compilation.options,
+                    compilation,
+                    compilation.resolverFactory.get('normal'),
+                    compilation.compiler.inputFileSystem,
+                    () => (syncCheck = true)
+                );
+                if (!syncCheck) {
+                    throw new Error('Expect CssModule build to be sync');
```

Comment:
Why is this validation needed? Seems oddly specific and is not promised by the API...
Can't we build it asynchronously if needed?

Reply:
I wanted to keep things small. it will cause a greater change the hook that we use is a syncHook and I will need to find a better way to solve it. so for now at least I validate the assumption.